controller/controller.go

```diff
@@ -1437,6 +1437,9 @@ func (ctrl *ProvisionController) provisionClaimOperation(ctx context.Context, cl
 	glog.Info(logOperation(operation, "succeeded"))
 
 	if err := ctrl.volumeStore.StoreVolume(claim, volume); err != nil {
+		if volumAddErr := ctrl.volumes.Add(volume.Name); volumAddErr != nil {
```

Comment:
You should add the volume to cache on success, not after failure.

Reply:
Fixed! I have no idea how it got in there